apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -122,24 +113,32 @@ where
         let enable_authorization_directives =
             AuthorizationPlugin::enable_directives(configuration, &schema).unwrap_or(false);
 
-        let config_mode = match configuration.experimental_query_planner_mode {
+        let mut hasher = StructHasher::new();
+        match configuration.experimental_query_planner_mode {
```

Comment:
If we hashing different structures we need to add some prefix otherwise we can have collision where values are matching in different structures (note: property names are not hashed only values).

Reply:
done in df8e6fc32